src/Test/L0/ProcessExtensionL0.cs

```diff
@@ -49,30 +49,37 @@ public async Task SuccessReadProcessEnv()
 
                     try
                     {
-                        await Task.Delay(2000);
                         trace.Info($"Read env from {timeout.Id}");
-                        var value = timeout.GetEnvironmentVariable(hc, envName);
-                        if (string.Equals(value, envValue, StringComparison.OrdinalIgnoreCase))
+                        int retries = 5;
+                        while (retries > 0)
                         {
-                            trace.Info($"Find the env.");
-                            return;
-                        }
-                        else
-                        {
-                            Assert.True(false, "Expected environment '" + envValue + "' did not match actual '" + value + "'");
+                            try
+                            {
+                                var value = timeout.GetEnvironmentVariable(hc, envName);
+                                Assert.True(string.Equals(value, envValue, StringComparison.OrdinalIgnoreCase), "Expected environment '" + envValue + "' did not match actual '" + value + "'");
+                                break;
+                            }
+                            catch (Exception ex)
+                            {
+                                retries--;
```

Comment:
Do we have off by one here? If I set retries to 1 and it throws I decrement to 0 and the code throws before doing the retry. Check goes before decrement?

Reply:
I think I was thinking that if you set retries to 1, you only wanted to try once, but I could get behind retries==1 meaning, 1 try + 1 retry. I'll update.